### ISSUE TITLE
[Tag] Improve accessibility of the removable tag

### ISSUE THREAD

**Issue by anna-lach** [sl-design-system/components#2845]
### Provide a general summary of the issue here

The removable `tag` right now can be removed when pressing `Backspace`/`Delete` key, but it should be possible to remove it with `Space`/`Enter` key instead (so just like normal action on a button). It would be more clear for the assistive technology users, they will understand it's an action. 

We probably won't need the `aria-describedby` there anymore (?)

### 👤 Your name

Anna Sobczak

### 🧢 Your Product/Team

SLDS

**Comment by a11ymiko** [comment]
This is fixed by: #3214
